presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/mysql/MySqlFunctionNamespaceManager.java

```diff
@@ -247,12 +247,16 @@ public void dropFunction(QualifiedObjectName functionName, Optional<List<TypeSig
             FunctionNamespaceDao transactionDao = handle.attach(functionNamespaceDaoClass);
             List<SqlInvokedFunction> functions = getSqlFunctions(transactionDao, functionName, parameterTypes);
 
-            checkUnique(functions, functionName);
             checkExists(functions, functionName, parameterTypes);
 
-            SqlInvokedFunction latest = functions.get(0);
-            checkState(latest.getVersion().isPresent(), "Function version missing: %s", latest.getFunctionId());
-            transactionDao.setDeletionStatus(hash(latest.getFunctionId()), latest.getFunctionId(), getLongVersion(latest), true);
+            if (functions.size() > 1) {
+                transactionDao.setDeleted(functionName.getCatalogName(), functionName.getSchemaName(), functionName.getObjectName());
```

Comment:
Are these overload versions? If so, please fix the title to reflect that.

Reply:
Yes, and updated.